plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCShippingLabelStore.kt

```
@@ -453,4 +498,23 @@ class WCShippingLabelStore @Inject constructor(
             }
         }
     }
+
+    suspend fun createPackages(
+        site: SiteModel,
+        customPackages: List<CustomPackage> = emptyList(),
+        predefinedPackages: List<PredefinedOption> = emptyList()): WooResult<Boolean> {
+        // We need at least one of the lists to not be empty to continue with API call.
+        if (customPackages.isEmpty() && predefinedPackages.isEmpty()) {
+            return WooResult(WooError(GENERIC_ERROR, UNKNOWN))
```

A `WooResult` here will make the consumers unaware of the cause, generally when it's an issue with arguments that we don't want to allow, we would throw an `IllegalArgumentException` with a clear message of what's happening, this way the consumer would fail early, and any hidden bug will be caught during the development phase.

---

Oh good idea, I was wondering how to do this properly too. I'll work on that.